### ISSUE TITLE
Effect Chain Wet/Dry

### ISSUE THREAD

**Issue by Spekular** [LMMS/lmms#4403]
The ability to mix between a dry signal and a signal affected by several effects would be useful. Currently this would require two mixer channels at minimum.

If this were implemented as a horizontal slider it wouldn't necessarily take too much real estate. In the English translation, there is space next to the "effects chain" header, for example. Otherwise it could be placed on a row by itself to avoid potential issues with translation.

**Comment by SecondFlight** [comment]
I personally like the idea of a toggle better. I don't see a solid use-case for putting all your effects at halfway when there is already a way to fine-tune them individually. I do have a bias here since the software I use does it this way.

If you want the same effect as putting a channel at halfway, you could route your input to a different channel, route that channel to two separate channels, and put your effects on one and not the other. Then you'd have a wet and a dry signal to mix in. I realize this is a somewhat roundabout way to do it.

**Comment by musikBear** [comment]
> I don't see a solid use-case for putting all your effects at halfway when there is already a way to fine-tune them individually.

Its as automation the usecase exists. The DRY/WET amount on each of ..say four different effects are *different* from each other, and Specular' slider would fade the *chain* as a whole. This is quite often seen on pro-mixer-tables : proportional slider changes. I like the idea of being able to fade an effect *chain* in/out, but there is issues with what to do, if one DW dials goes < 0 or >max, as the chain-slider changes -A similar problem as in AFPs loop-settings.

**Comment by SecondFlight** [comment]
Well no, something like this would be implemented separately from the individual wet/dry mix knobs. Connecting them all together would create all sorts of problems and unnecessary complexity.

The question in my head is, is it worth the extra complexity in the UI to save the 30 seconds it would take to link 4 different automation clips instead of one. This would need to be used quite often to merit this kind of extra complexity IMO, and I don't think it would be.

I recognize that this is an opinion, and at the end of the day it would appear I'm in the minority. I would welcome anyone else to add their thoughts.

At the end of the day, if one of the devs sees it as a good enough idea to go to the work of implementing it then I won't complain.

**Comment by Spekular** [comment]
Automating the W/D of the channels separately isn't the same thing as a universal W/D though, it'll give a different sound.

**Comment by SecondFlight** [comment]
I suppose it would in some cases. But then going back to the alternative, can't you accomplish the same thing now with relative ease? That's what channel sends are for.

**Comment by Spekular** [comment]
If the effects are placed on the instrument itself, you'd need to clone that instrument to get a clean signal, and keep the cloned tracks in sync. You could then send both these to the same FX channel to get a 50/50 mix, but you'd need to adjust their volumes to change the W/D mix. If something like a hard limiter is used in your effects chain, you'd have to use an amplifier for this volume adjustment, further complicating things.

If the effects are placed on a mixer channel (or moved there from an instrument, which is a time consuming process right now) you need a minimum of two mixer channels for this instrument. The simplest layout would be sending your instrument to a clean channel, which then sends to master and a wet channel.

In either of these scenarios, you need two separate automation tracks to adjust the W/D.

With a W/D adjust, your mixed signal sends from one channel, which means you only need to set sends and send amounts on one channel, and soloing that channel will work correctly. To accomplish this without W/D you'd need an additional channel to sum your wet and dry channels.

In essence, the mixer approach is messier and more work. This is fine if you're only adjusting the W/D on a small number of instruments or in a simple manner. If you have several chains to adjust, especially if they feed into each other, it becomes much more of a burden.

Personally I think convenience features are very valuable. The time and effort they save adds up & lets you focus on more important aspects of production.

**Comment by SecondFlight** [comment]
Yeah, I see where you're coming from. I think my objection comes down to the fact that I would never use this feature, but that's not to say nobody else would.

**Comment by tresf** [comment]
As part of a pruning effort, this `enhancement` request is archived into a dedicated "Better Workflow" checklist here #4877.
